File: string/415-checkPrefixAndSuffix.py


def isPrefix(p,s):
    if len(p)>len(s):
        return False
    i=0
    while(i<len(p)):
        if p[i]!=s[i]:
            return False
        i+=1
    return True

def isSuffix(p,s):
    if len(p)>len(s):
            return False
    ip,iss=len(p)-1,len(s)-1
    while(ip>=0):
        if p[ip]!=s[iss]:
            return False
        ip-=1
        iss-=1
    return True


p="abcdE"
s="abcdefgabcd"
print(isPrefix(p,s))
print(isSuffix(p,s))
    

File: string/415-checkPrefixAndSuffix.cpp
#include<iostream>
#include<string>

using STR=std::string;

bool isPrefix(const STR& p,const STR& s)
{
    if(p.size()>s.size())
        return false;
    
    int i=0;
    while(i<p.size())
    {
        if(p[i]!=s[i])
            return false;
        i+=1;
    }
    return true;
}


bool isSuffix(const STR& p, const STR& s)
{
    if(p.size()>s.size())
        return false;
    
    int i=p.size()-1,j=s.size()-1;
    while(i>=0)
    {
        if(p[i]!=s[j])
            return false;
        i-=1;
        j-=1;
    }
    return true;
}

int main()
{
    STR p{"abcd"},s{"abcdefgabcd"};
    std::cout<<isPrefix(p,s);
    std::cout<<isSuffix(p,s);
    return 0;
}